Mathlib/Tactic/Bound/Attribute.lean

```diff
@@ -0,0 +1,103 @@
+/-
+Copyright (c) 2024 Geoffrey Irving. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Geoffrey Irving
+-/
+
+import Aesop.Frontend.Attribute
+import Mathlib.Tactic.Bound.Init
+
+/-!
+# The `bound` attribute
+
+Any lemma tagged with `@[bound]` is registered as an apply rule for the `bound` tactic, by
+converting it to either `norm apply` or `safe apply <priority>`.  The classification is based
+on the number and types of the lemmas hypotheses.
+-/
+
+namespace Bound
+
+initialize Lean.registerTraceClass `bound.attribute
+
+/-- Check if an expression is zero -/
+def isZero : Lean.Expr → Bool
+  | .lit (.natVal 0) => true
+  | .const `Zero.zero _ => true
+  | .app (.app (.app (.const `OfNat.ofNat _) _) z) _ => isZero z
+  | _ => false
```

Comment:
This incorrectly handles meta-data: If I pass in `Zero.zero` wrapped in `mdata`, this will incorrectly say it is not zero. You should use Qq matching instead.

Reply:
Can you look at what `positivity` does? There should be the same problem there.

Reply:
Done.